src/main/java/org/opensearch/geospatial/ip2geo/jobscheduler/Datasource.java

```diff
@@ -0,0 +1,487 @@
+        /**
+         * @param validForInDays A duration in which the database file is valid to use
+         * @return A duration in which the database file is valid to use
+         */
+        private Long validForInDays;
+        /**
+         * @param fields A list of available fields in the database
+         * @return A list of available fields in the database
+         */
+        private List<String> fields;
+
+        private static final ConstructingObjectParser<Database, Void> PARSER = new ConstructingObjectParser<>(
+            "datasource_metadata_database",
+            true,
+            args -> {
+                String provider = (String) args[0];
+                String md5Hash = (String) args[1];
+                Instant updatedAt = args[2] == null ? null : Instant.ofEpochMilli((Long) args[2]);
+                Long validForInDays = (Long) args[3];
+                List<String> fields = (List<String>) args[4];
+                return new Database(provider, md5Hash, updatedAt, validForInDays, fields);
+            }
+        );
+        static {
+            PARSER.declareString(ConstructingObjectParser.optionalConstructorArg(), PROVIDER_FIELD);
+            PARSER.declareString(ConstructingObjectParser.optionalConstructorArg(), MD5_HASH_FIELD);
+            PARSER.declareLong(ConstructingObjectParser.optionalConstructorArg(), UPDATED_AT_FIELD);
+            PARSER.declareLong(ConstructingObjectParser.optionalConstructorArg(), VALID_FOR_IN_DAYS_FIELD);
+            PARSER.declareStringArray(ConstructingObjectParser.optionalConstructorArg(), FIELDS_FIELD);
+        }
+
+        @Override
+        public XContentBuilder toXContent(final XContentBuilder builder, final Params params) throws IOException {
+            builder.startObject();
+            if (provider != null) {
+                builder.field(PROVIDER_FIELD.getPreferredName(), provider);
+            }
+            if (md5Hash != null) {
+                builder.field(MD5_HASH_FIELD.getPreferredName(), md5Hash);
+            }
+            if (updatedAt != null) {
+                builder.timeField(
+                    UPDATED_AT_FIELD.getPreferredName(),
+                    UPDATED_AT_FIELD_READABLE.getPreferredName(),
+                    updatedAt.toEpochMilli()
+                );
+            }
+            if (validForInDays != null) {
+                builder.field(VALID_FOR_IN_DAYS_FIELD.getPreferredName(), validForInDays);
+            }
+            if (fields != null) {
+                builder.startArray(FIELDS_FIELD.getPreferredName());
+                for (String field : fields) {
+                    builder.value(field);
+                }
+                builder.endArray();
+            }
+            builder.endObject();
+            return builder;
+        }
+    }
+
+    /**
+     * Update stats of a datasource
+     */
+    @Getter
+    @Setter
+    @AllArgsConstructor
+    public static class UpdateStats implements ToXContent {
```

Comment:
Can we move all the inner classes UpdateStats, Database outside of this class? and if they are only required in this class, may be you want to make them package private.

Reply:
@heemin32

Reply:
Discussed offline. 

Datasource is a data model. This class is inner object of parent object in json representation. Therefore, better to keep it inside here.

Could not make it `private` as that will block the access of the class outside of parent class. Instead, made all of its constructor as private so that the object cannot be created outside of parent class.